src/map/script.cpp

```diff
@@ -20309,27 +20309,62 @@ BUILDIN_FUNC(instance_check_clan)
 /*==========================================
 * instance_info
 * Values:
-* name : name of the instance you want to look up. [Required Parameter]
 * type : type of information you want to look up for the specified instance. [Required Parameter]
+* name/ID : name or instance ID of the instance you want to look up. [Optional Parameter]
 * index : index of the map in the instance. [Optional Parameter]
 *------------------------------------------*/
 BUILDIN_FUNC(instance_info)
 {
-	const char* name = script_getstr(st, 2);
-	int type = script_getnum(st, 3);
-	int index = 0;
-	struct instance_db *db = instance_searchname_db(name);
+	struct script_data *data;
+	struct instance_db *db = NULL;
+	struct instance_data *im = NULL;
+	const char* name = NULL;
+
+	int type = script_getnum(st, 2);
+	int index = 0, id = 0;
 
+	if( !script_hasdata(st, 3) ) {
+		id = script_instancegetid(st);
+		if( id ) {
+			im = &instance_data[id];
+			db = instance_searchtype_db(im->type);
+		}
+	}
+	else {
+		data = script_getdata(st, 3);
+		get_val(st, data);
+		if( data_isstring(data) ) {
+			name = script_getstr(st, 3);
+			db = instance_searchname_db(name);
+		}
+		else if( data_isint(data) || script_getnum(st, 3) ) {
```

Comment:
I would only treat this as "else" case or is there anything else possible than string or number?